portsyncd/linksync.cpp

```diff
@@ -95,20 +95,27 @@ LinkSync::LinkSync(DBConnector *appl_db, DBConnector *state_db) :
     if (!WarmStart::isWarmStart())
     {
         /* See the comments for g_portSet in portsyncd.cpp */
-        for (string port : g_portSet)
+        for (auto port_iter = g_portSet.begin(); port_iter != g_portSet.end();)
         {
+            string port = *port_iter;
             vector<FieldValueTuple> temp;
+            bool portFound = false;
             if (m_portTable.get(port, temp))
             {
                 for (auto it : temp)
                 {
                     if (fvField(it) == "admin_status")
                     {
-                        g_portSet.erase(port);
+                        port_iter = g_portSet.erase(port_iter);
+                        portFound = true;
                         break;
                     }
                 }
             }
+            if (!portFound)
+            {
+                ++port_iter;
+            }
```

Comment:
Erasing from a set while iterating is UB, we haven't seen this bug before because this code was never executed before because of LinkSync and g_portSet initialization order. Please check below comment regarding LinkSync.

Reply:
thanks for fixing this issue! could you help move this part of the code as a separate pull request? we could merge this bug fix and the feature one by one.

Reply:
Please take a look https://github.com/Azure/sonic-swss/pull/1019
Once this is merged I will rebase this PR

Reply:
done